app/services/logging/logger.spec.js

```diff
@@ -0,0 +1,62 @@
+"use strict";
+
+const fs = require("fs");
+const Logger = require("./logger");
+const LoggerConfig = require("./loggerConfig");
+const Winston = require("winston");
+require("winston-daily-rotate-file");
+
+describe("logger", () => {
+  it("does not create a Winston instance if transports is undefined", function () {
+    const loggerConfig = new LoggerConfig({ appConfig: {} });
```

Comment:
By instantiating the LoggerConfig, you are potentially polluting your tests with logic that doesn't concern them. I appreciate that LoggerConfig is just a POJO (plain old javascript object - i.e. one that has only properties and does not do anything complex), but because JS isn't strongly typed, you can pass this object as a fake then populate *only* the properties that your test is concerned with. This also means that if the LoggerConfig object changes its behaviour, your logger class also needs to be updated and tests may fail when they shouldn't.

See https://github.com/lewster32/corporallancot/blob/master/app/services/db/adapters/mariaDbAdapter.spec.js for examples.

You can also use faker.js to easily create fake objects / properties for more confident testing (that doesn't rely on particular hard coded values): https://github.com/lewster32/corporallancot/blob/master/app/actions/persistenceHandlers/notesActionPersistenceHandler.spec.js

Reply:
I think if you did this you'd also see less tests and / or more meaningful tests.